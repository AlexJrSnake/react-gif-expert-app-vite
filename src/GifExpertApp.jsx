import React, { useState } from 'react'
import { GifGrid, AddCategory } from './components'

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['HunterXHunter'])

    const onAddCategory = (NewValue) => {

        if (categories.includes(NewValue)) {
            return;
        }

        setCategories([NewValue,...categories])
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory 
                onNewCategory={onAddCategory}
            />
            { 
                categories.map( (element, index) => {
                    return <GifGrid key={element} category={element}></GifGrid>
                }) 
            }            
        </>
    )


}
