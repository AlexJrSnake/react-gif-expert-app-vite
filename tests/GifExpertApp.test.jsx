import { fireEvent, render, screen } from "@testing-library/react"
import {GifExpertApp} from "../src/GifExpertApp"

describe('<GifExpertApp></GifExpertApp>', () => { 


    test('Snapshot', () => {
        const {container} = render(<GifExpertApp/>)
        expect(container).toMatchSnapshot()
    })


    test('Buscar Girf Value', () => {
        const inputValue = 'Saitama'
        render(<GifExpertApp/>)
        const input = screen.getByRole('textbox'); 
        fireEvent.input( input , { target: {value: inputValue}} )
        expect( input.value ).toBe('Saitama')
    })

})