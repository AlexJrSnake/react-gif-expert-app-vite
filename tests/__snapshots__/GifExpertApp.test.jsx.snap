// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp></GifExpertApp> Snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Girf"
      type="text"
      value=""
    />
  </form>
  <h3>
    HunterXHunter
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
